plugins/modules/foreman_host.py

```diff
@@ -85,76 +115,57 @@
 
 RETURN = ''' # '''
 
-try:
-    from nailgun.entities import (
-        Host,
-        HostGroup,
-    )
-
-    from ansible.module_utils.ansible_nailgun_cement import (
-        find_host,
-        find_hostgroup,
-        find_location,
-        find_organization,
-        ForemanEntityAnsibleModule,
-        naildown_entity_state,
-        sanitize_entity_dict,
-    )
-except ImportError:
-    pass
-
-
-# This is the only true source for names (and conversions thereof)
-name_map = {
-    'name': 'name',
-    'enabled': 'enabled',
-    'managed': 'managed',
-    'hostgroup': 'hostgroup',
-    'location': 'location',
-    'organization': 'organization',
-}
+from ansible.module_utils.foreman_helper import ForemanEntityAnsibleModule
 
 
 def main():
     module = ForemanEntityAnsibleModule(
-        argument_spec=dict(
+        entity_spec=dict(
             name=dict(required=True),
-            hostgroup=dict(),
-            location=dict(),
-            organization=dict(),
-            enabled=dict(default='true', type='bool'),
+            hostgroup=dict(type='entity', flat_name='hostgroup_id'),
+            location=dict(type='entity', flat_name='location_id'),
+            organization=dict(type='entity', flat_name='organization_id'),
+            enabled=dict(type='bool'),
             managed=dict(type='bool'),
-            state=dict(default='present', choices=[
-                       'present_with_defaults', 'present', 'absent']),
+            build=dict(type='bool'),
         ),
         required_if=(
-            ['state', 'present_with_defaults', ['hostgroup']],
-            ['state', 'present', ['hostgroup']],
+            ['managed', True, ['hostgroup']],
+            ['build', True, ['hostgroup']],
         ),
     )
 
-    (host_dict, state) = module.parse_params()
+    entity_dict = module.clean_params()
 
-    module.connect()
+    # additional param validation
```

Comment:
this should add the "is the hostname a FQDN" check we discussed
```suggestion
    # additional param validation
    if '.' not in entity_dict['name']:
        module.fail_json(msg="The host name must include the domain")
```

Reply:
silly check to my taste, but we can't really do any more yet. Done

Reply:
Silly because it's just checking for a dot? Yeah. But better than nothing :)
Thanks!